courses/views_test.py

```diff
@@ -0,0 +1,188 @@
+"""
+Tests for course views
+"""
+import pytest
+from django.urls import reverse
+
+from mitxpro.test_utils import JSON_CONTENT_TYPE_PARAM
+from courses.factories import ProgramFactory, CourseFactory, CourseRunFactory
+from courses.models import Program, Course, CourseRun
+from courses.serializers import ProgramSerializer, CourseSerializer, CourseRunSerializer
+
+
+pytestmark = [pytest.mark.django_db]
+
+
+class TestProgramViews:  # pylint: disable=missing-docstring
+    @pytest.fixture()
+    def programs(self):
+        """Fixture for a set of Programs in the database"""
+        return ProgramFactory.create_batch(3)
+
+    def test_get_programs(self, user_client, programs):
+        """Test the view that handles requests for all Programs"""
+        resp = user_client.get(reverse("programs_api-list"))
+        programs_data = resp.json()
+        assert len(programs_data) == len(programs)
+        for program, program_data in zip(programs, programs_data):
+            assert program_data == ProgramSerializer(program).data
+
+    def test_get_program(self, user_client, programs):
+        """Test the view that handles a request for single Program"""
+        program = programs[0]
+        resp = user_client.get(
+            reverse("programs_api-detail", kwargs={"pk": program.id})
+        )
+        program_data = resp.json()
+        assert program_data == ProgramSerializer(program).data
+
+    def test_create_program(self, user_client, programs):
+        """Test the view that handles a request to create a Program"""
+        program = programs[0]
+        program_data = ProgramSerializer(program).data
```

Comment:
I think this should be changed to an explicit `dict` with params. Since the view tests here serve as a test of the serializers, this is basically using the serializer to test itself so I worry about something not getting caught because it's basically a closed loop.

Reply:
I can just add separate tests for serializers. I initially left that out since the serializers are based on the most basic DRF usage